package/Makefile

```diff
@@ -86,6 +87,13 @@ $(curdir)/index: FORCE
 	@for d in $(PACKAGE_SUBDIRS); do ( \
 		mkdir -p $$d; \
 		cd $$d || continue; \
+		$(STAGING_DIR_HOSTPKG)/bin/apk mkndx \
+			--root $(TOPDIR) \
+			--keys-dir $(TOPDIR) \
+			--sign $(BUILD_KEY_APK_SEC) \
+			--output APKINDEX.tar.gz \
+			*.apk; \
+		ln -sf . $(ARCH); \
```

Comment:
This is currently awkward, `apk` always attaches the `arch` to whatever repository you hand it. OpenWrt follow the structure `packages/<arch>/<feed>` while `apk` uses `/packages/<feed>/<arch>`. We can either [patch `apk`](https://github.com/alpinelinux/apk-tools/blob/master/src/database.c#L594) or go with that structure. I don't think it would be an issue for the buildbots.

@jow- @dangowrt thoughts?

Reply:
It would be preferable if OpenWrt kept the standard URIs, as it would allow compatibility with other apk-related tooling, for example, you could target OpenWrt with `abuild rootbld`.

Reply:
@kaniini this isn't possible in a clean way, multiple targets offer target specific packages which are not shared between the architecture. The path is `/targets/<target>/<subtarget>/packages/`.

Reply:
Yeah I see what you’re saying.  I think then it’s best if apk3 just uses raw URLs.  I’ll try to make that case.